### ISSUE TITLE
NoMethodError: undefined method `next_token' for nil:NilClass

### ISSUE THREAD

**Issue by jlintz** [rodjek/puppet-lint#790]
---
puppet-lint version: 2.3.3
ruby version: 2.4.0-p0
platform: x86_64-darwin16
file path: /Users/justin/code/puppet/test.pp
file contents:
```
class test {
 file { '/tmp/foo':
  content => "
  test= '${::lookup('test', { 'default_value' => ''})}'
  "
  }
}

```
error:
```
NoMethodError: undefined method `next_token' for nil:NilClass
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/lexer/token.rb:196:in `find_token_of'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/lexer/token.rb:142:in `next_token_of'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/data.rb:176:in `block in resource_indexes'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/data.rb:171:in `each'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/data.rb:171:in `resource_indexes'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/checkplugin.rb:81:in `resource_indexes'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/plugins/check_conditionals/selector_inside_resource.rb:7:in `check'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/checkplugin.rb:21:in `run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/checks.rb:60:in `block in run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/checks.rb:57:in `each'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/checks.rb:57:in `run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint.rb:196:in `run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/bin.rb:59:in `block in run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/bin.rb:56:in `each'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/lib/puppet-lint/bin.rb:56:in `run'
/Users/justin/.rvm/gems/ruby-2.4.0/gems/puppet-lint-2.3.3/bin/puppet-lint:7:in `<top (required)>'
/Users/justin/.rvm/gems/ruby-2.4.0/bin/puppet-lint:23:in `load'
/Users/justin/.rvm/gems/ruby-2.4.0/bin/puppet-lint:23:in `<main>'
/Users/justin/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `eval'
/Users/justin/.rvm/gems/ruby-2.4.0/bin/ruby_executable_hooks:15:in `<main>'

**Comment by Hufschmidt** [comment]
```puppet
class profile::database::postgresql {
  @nagios::check { 'pgsql':
    content => "check_pgsql -P ${lookup('postgresql::server::port', { value_type => Integer, default_value => 5432 })}"
  }
}
```
Having a similar error when using lookup-function inside string-interpolation.

**Comment by keur** [comment]
puppet-lint version: 2.3.6
ruby version: 2.3.3-p222
platform: x86_64-linux-gnu
file path: modules/ocf_mail/manifests/init.pp
file contents:
```
class ocf_mail {
  ...
  ...
  ocf::systemuser { 'ocfmail':
    opts    => {
      ensure => present,
      name   => ocfmail,
      gid    => ocfmail,
      home   => '/var/mail',
      shell  => '/bin/false',
    },
    require => Group['ocfmail'],
  }

  group { 'ocfmail':
    ensure => present,
    name   => ocfmail,
    system => true,
  }
  ...
  ...
}

```
error:
```
NoMethodError: undefined method `next_token_of' for nil:NilClass
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/plugins/check_resources/ensure_first_param.rb:32:in `fix'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:42:in `block in fix_problems'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:38:in `each'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checkplugin.rb:38:in `fix_problems'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:67:in `block in run'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:65:in `each'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/checks.rb:65:in `run'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint.rb:205:in `run'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:66:in `block in run'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `each'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/lib/puppet-lint/bin.rb:62:in `run'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/gems/puppet-lint-2.3.6/bin/puppet-lint:7:in `<top (required)>'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/bin/puppet-lint:22:in `load'
/home/k/ke/keur/.cache/pre-commit/reponi64q5zn/rbenv-default/gems/bin/puppet-lint:22:in `<main>'
```

Same issue here, I think. For my case, it errors out because `ensure => present` is not the first line. Moving it there doesn't output an error.

**Comment by rodjek** [comment]
Fixed by #846
